dtrace: add page (#10787)
tldr-pages/tldr: pages/osx/dtrace.md

# dtrace

> > Generic front-end to DTrace facility.
> > This command is a simple interface to invoke the D language compiler, retrieve buffered trace data from the DTrace kernel facility, and a set of basic routines to format and print traced data.
> > This requires root privileges.

Reviewer: ```suggestion
> A simple interface to invoke the D language compiler, retrieve buffered trace, and print traced data from the DTrace kernel facility.
> Generic front-end to DTrace facility, requiring root privileges.
```

Author: Done.

> More information: <https://www.unix.com/man-page/osx/1/dtrace/>.

> - Set dtrace's target data model which is the target architecture:

Reviewer: ```suggestion
- Set target data model for a specific architecture:
```

Author: Done.

`dtrace -arch {{arch_name}}`

> - Claim anonymous tracing state and display the traced data:

Reviewer: ```suggestion
- Claim [a]nonymous tracing state and display the traced data:
```

Author: Done, but why is this necessary?

Reviewer 2: Mnemonics are used interchangeably to highlight long and short options in example descriptions.

`dtrace -a`

> - Set principal trace buffer size. Trace buffer size can include any size suffix like k, m, g or t:
> 
> `dtrace -b {{buffer_size}}`

Reviewer: ```suggestion
- Set principal trace buffer size. Supported units are `k`, `m`, `g`, or `t`:

`dtrace -b {{2g}}`
```

Reviewer 2: Is `2g` as buffer size a typo or is it a supported unit too?

Reviewer: It's just an example of 2 gb.

Author: Done

> - Compile the specified D Program:

Reviewer: ```suggestion
- Compile the specified D Program [s]ource file:
```

Author: Done

`dtrace -s {{D_script}}`

> - Run tghe specified command and exit upon its completion:
> 
> `dtrace -c {{path/to/command}}`

Reviewer: ```suggestion
- Run the specified [c]ommand and exit upon its completion:

`dtrace -c {{command}}`
```

Author: Done

> - Specify function name to trace or list (-l option).  The corresponding argument can include any of the probe description  forms like provider:module:function, module:function or function:

Reviewer: ```suggestion
- Specify [f]unction name to trace or list (-l option). The corresponding argument can include any of the probe description forms like `provider:module:function`, `module:function` or `function`:
```

Author: Done

`dtrace -f {{function}}`

> - Grad the process corresponding to specified pid , cache its symbol table and exit upon completion:

Reviewer: ```suggestion
- Grad the specified  [p]rocess ID, cache its symbol table, and exit upon completion:
```

Author: Done
